pkg/check/cluster.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2025 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package check
+
+import (
+	"context"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/pingcap/log"
+	cerror "github.com/pingcap/tiflow/pkg/errors"
+	pmysql "github.com/pingcap/tiflow/pkg/sink/mysql"
+	pd "github.com/tikv/pd/client"
+	"go.uber.org/zap"
+)
+
+var (
+	dbConnImpl  pmysql.IDBConnectionFactory = &pmysql.DBConnectionFactory{}
+	checkIsTiDB                             = pmysql.CheckIsTiDB
+	// Export for testing only.
+	GetClusterIDBySinkURIFn = getClusterIDBySinkURI
```

Comment:
Please make this var private and add a public method name `SetGetClusterIDBySinkURIFnForTest()` for setting it to prevent mistakenly setting.

Reply:
Good idea.